keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,487 @@
+* Try to modify the reserved label and this should be a no-op- the label is guaranteed to be there.
+* Check the WATCH operation against existing namespaces that do not have this label still work properly.
+
+### Graduation Criteria
+* This feature being enabled by default at least one release.
+
+### Upgrade / Downgrade Strategy
+
+* Check the WATCH situation described in test plan, otherwise populate the label 
+of namespaces without it.
+* In case of downgrade, as this is just a label it will work fine. Network Policies 
+that rely on this label w.r.t. their `matchLabels` criteria within namespaceSelector
+to enforce rules will fail-close, as they no longer will match these namespaces,
+thereby fail to add such namespaces to their whitelist rules.
+On the other hand, network policies with `matchExpressions` within a
+namespaceSelector, specifically with a `notIn` operator as shown below, will
+inadvertently select this namespace instead of excluding it, which is an undesired
+side effect.
+
+```
+namespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: NotIn
+    values:
+    - kube-system
+```
+
+Hence, it must be noted that namespaces are not guaranteed to have this label on
+downgrades to release which do not support this feature. Users may choose to label
+their namespaces by themselves in order to restore their expected behaviour in
+terms of network policy rules.
+
+### Version Skew Strategy
+
+It is assumed that losing data under a `kubernetes.io` label is acceptable, since its a reserved label and should not generally be used by non-k8s core components:
+
+```
+The kubernetes.io/ and k8s.io/ prefixes are reserved for Kubernetes core components.
+```
+
+which is published in https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/.
+
+Since this has been reserved since 2017 (if not earlier), no backwards guarantees are made for clusters using the `kubernetes.io/metadata.name` label, but we'll do our best to not break people who coincidentally may have used this label (debatable wether this is a realistic concern or not).
+
+<<[UNRESOLVED Very likely to have a warning release, but not 100% sure ]>>
+
+- There is an ongoing debate, @thockin, @liggit, about wether a 'buffer' release is needed before this label is stomped.
+- Old versions will be assumed to never have this label.  If they do have this label, then they will eventually lose its contents.
+- New versions always will have this label, lazily added on read or write.
+- During upgrades, a policy created may be able to "get away" with temporarily writing to the aformentioned "kubernetes.io" metadata.name label, however, this write will eventually be lost.
+
+<<[/UNRESOLVED]>>
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+1.21:
+- gate disabled by default
+- if gate disabled, log or event incompatible uses of this
```

Comment:
@liggitt if we are unilaterally stomping, and the entire net change is 6 LOC + tests, do we even need an alpha cycle?

Reply:
starting in beta, enabled by default, unilaterally stomping sounds good to me

Reply:
@liggitt do we want to merge and iterate, or do you want me to make more changes now ... ?

Reply:
I'm ok with a follow-up dropping the warning release, starting in beta, and resolving the unresolved sections before moving to implementable

/lgtm
/approve

(needs an api-machinery approver)

Reply:
ping @deads2k ^

Reply:
@liggitt what does "beta" mean for this?  Gate-disableable?

Reply:
That was my thought, yeah. A one release grace period to opt out of the behavior at the cluster deployer level if needed in 1.21.

Reply:
gate-disableable for one release SGTM

Reply:
pushed thanks ! yeah, removed 1.23